src/HTMLMutator.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import nullthrows from 'nullthrows';
+import uuid from './ui/uuid';
+
+// Utility Class that allows developer to insert HTML snippets then updates
+// document's innerHTML accordingly.
+export default class HTMLMutator {
+  _doc: Document;
+  _htmls: Map<string, string>;
+
+  constructor(doc: Document) {
+    this._doc = doc;
+    this._htmls = new Map();
+  }
+
+  insertHTMLBefore(html: string, el: Element): HTMLMutator {
+    return this._insertHTML(html, 'before', el);
+  }
+
+  insertHTMLAfter(html: string, el: Element): HTMLMutator {
+    return this._insertHTML(html, 'after', el);
+  }
+
+  execute(): void {
+    const doc = this._doc;
+    const root = nullthrows(doc.body || doc.documentElement);
+    let newHtml = root.innerHTML;
+    this._htmls.forEach((html, token) => {
```

Comment:
`token` feels a little misleading here, because that second param is an index, right?